File: CodingTest/Coding_Test/programmers/2021 카카오 채용연계형 인턴십/거리두기 확인하기.java
import java.util.*;

class Solution {

    public static boolean check(String[][] board,List<int[]> people){

        int numOfPeople=people.size();
        for(int i=0;i<numOfPeople-1;i++){
            int x1=people.get(i)[0];
            int y1=people.get(i)[1];
            for(int j=i+1;j<numOfPeople;j++){
                int x2=people.get(j)[0];
                int y2=people.get(j)[1];
                int dist=Math.abs(x1-x2)+Math.abs(y1-y2);
                if(dist>2){
                    continue;
                }else if(dist<2){
                    return false;
                }else if(dist==2){
                    if((x1-x2)==0){
                        if(board[x1][(y1+y2)/2].equals("X")){
                            continue;
                        }else{
                            return false;
                        }
                    }else if((y1-y2)==0){
                        if(board[(x1+x2)/2][y1].equals("X")){
                            continue;
                        }else{
                            return false;
                        }
                    }else if(Math.abs(x1-x2)==1&&Math.abs(y1-y2)==1){
                        if(board[x1][y2].equals("X")&&board[x2][y1].equals("X")){
                            continue;
                        }else{
                            return false;
                        }
                    }
                }
            }
        }

        return true;
    }

    public int[] solution(String[][] places) {
        int[] answer = new int[5];

        for(int tc=0;tc<5;tc++){
            String[] k=places[tc];
            String[][] board=new String[5][5];
            List<int[]> people=new ArrayList<>();
            for(int i=0;i<5;i++){
                String[] tmp=k[i].split("");
                for(int j=0;j<5;j++){
                    if(tmp[j].equals("P")){
                        people.add(new int[]{i,j});
                    }
                }
                board[i]=tmp;
            }
            if(check(board,people)){
                answer[tc]=1;
            }
        }

        return answer;
    }
}

File: CodingTest/Coding_Test/programmers/2021 카카오 채용연계형 인턴십/거리두기 확인하기.py
def check_board(board,people):
  # print(board,people)
  numOfPeople=len(people)
  tf=True
  for i in range(0,numOfPeople-1):
      for j in range(i+1,numOfPeople):
          x1,y1=people[i]
          x2,y2=people[j]
          dist=abs(x1-x2)+abs(y1-y2)
          if dist>2:
              continue
          else:
              # 가로와 세로로 먼 경우
              if abs(x1-x2)==0 or abs(y1-y2)==0:
                  if abs(x1-x2)==0 and abs(y1-y2)<2:
                      return False
                  elif abs(x1-x2)==0 and abs(y1-y2)==2:
                      if board[x1][(y1+y2)//2]=='X':
                          continue
                      else:
                          return False
                  elif abs(y1-y2)==0 and abs(x1-x2)<2:
                      return False
                  elif abs(y1-y2)==0 and abs(x1-x2)==2:
                      if board[(x1+x2)//2][y1]=='X':
                          continue
                      else:
                          return False
              elif abs(x1-x2)==1 and abs(y1-y2)==1:
                  if board[x1][y2]=='X' and board[x2][y1]=='X':
                      continue
                  else:
                      return False
              else:
                  return False

  return tf

def solution(places):
  answer = []

  for tc in range(5):
      case=places[tc]
      board=[]
      people=[]
      for i in range(5):
          tmp=case[i]
          for j in range(5):
              if tmp[j]=='P':
                  people.append((i,j))
          board.append(list(tmp))
      if check_board(board,people):
          answer.append(1)
      else:
          answer.append(0)

  return answer